routes/ScannerExchangePage.vue

```diff
@@ -0,0 +1,417 @@
+              options.signOptions = {
+                profileId: options.verifiablePresentation.holder
+              };
+            }
+          }
+          exchanging.value = true;
+          const {value, done} = await exchange.value.next(options);
+          exchanging.value = false;
+          // clear share-related state
+          verifiablePresentationRequest.value = undefined;
+          verifiablePresentation.value = undefined;
+          if(value?.verifiablePresentation) {
+            // set store-related state
+            const {verifiablePresentation: presentation} = value;
+            holder.value = presentation.holder;
+            const {verifiableCredential: credentials} = presentation;
+            if(!credentials) {
+              verifiableCredential.value = [];
+            } else if(!Array.isArray(credentials)) {
+              verifiableCredential.value = [credentials];
+            } else {
+              verifiableCredential.value = credentials;
+            }
+            // if there is something to store, wait for user to approve storage
+            // FIXME: enable storage of capabilities as well
+            if(verifiableCredential.value.length > 0) {
+              // user must approve store...
+              setDisplay('store');
+              await wait();
+              actionTaken = true;
+            }
+            // clear store-related state
+            verifiableCredential.value = [];
+            holder.value = '';
+          }
+          if(value?.verifiablePresentationRequest) {
+            // user must approve share...
+            setDisplay('share');
+            // set share-related state
+            verifiablePresentationRequest.value =
+              value.verifiablePresentationRequest;
+            await wait();
+            actionTaken = true;
+          }
+          // if no action was taken then nothing actionable was in in `value`
+          // (no VPR, no VP / empty VP); so set an error condition and `wait()`
+          // to let the user see something went wrong
+          if(!actionTaken) {
+            const e = new Error('Nothing was received or requested.');
+            e.name = 'OperationError';
+            await wait();
+            throw e;
+          }
+          if(done) {
+            // exchange is finished
+            exchange.value.close();
+            $q.notify({
+              type: 'positive',
+              message: 'Successfully stored credential',
+            });
+            router.push({name: 'home'});
+            break;
+          }
+        }
+      } catch(e) {
+        console.error(e);
+        error.value = e;
+      } finally {
+        // ready.value = false;
+        exchanging.value = false;
+        // no exchange created; just reject event once user cancels
+        if(!exchange.value) {
+          event.respondWith(new Promise((res, rej) => {
+            reject.value = rej;
+          }));
+        }
+      }
+    }
```

Comment:
I'm assuming the above code is all the same as that other file, not closely reviewing it.

Reply:
This is old, good catch. Removing it in the next commit.